File: 511_library/Books.cpp
#include "Books.h"
#include <iostream>

using namespace std;


Books::Books(string call_number1,string title1,	string subjects1, string author1 ,
	string description1, string publisher1 , string city1 , string year1,
             string series1, string notes1):Media(call_number1, title1,subjects1, notes1)
{
     author = author1;
     description = description1;
     publisher = publisher1;
     city = city1;
     year = year1;
     series = series1;
}

void Books::display() const
{
	//int a;
	//cout<<"Call number"<<call_number<<endl;
	//cout<<"Title"<<title<<endl;
	//cout<<"Subjects"<<subjects<<endl;
	//cout<<"Notes"<<notes<<endl;
	cout<<"BOOK"<<endl;
	Media::display();
	cout<<"Author    :"<<author<<endl;
	cout<<"Description    :"<<description<<endl;
	cout<<"Publisher    :"<<publisher<<endl;
	cout<<"City   :"<<city<<endl;
	cout<<"Year    :"<<year<<endl;
	cout<<"Series   :"<<series<<endl;


	//cin>>a;
}

bool Books::compare_other(const string& ss)
{
	int i = -1;
	if (std::string::npos != notes.find(ss)||std::string::npos != description.find(ss)||std::string::npos != year.find(ss))
	{
		if (i >= 0)
			return true;
		else
			return false;
	}
	
	/*if (std::string::npos != description.find(ss))
	{
		if (i >= 0)
			return true;
		else
			return false;
	}

	if (std::string::npos != year.find(ss))
	{
		if (i >= 0)
			return true;
		else
			return false;
	}*/
}

File: Library_Python/Books.py
from Media import Media
class Books(Media): 
    def __init__(self,call_number1,title1,subjects1,author1,description1, publisher1,city1,year1,series1,notes1):
        super(Books, self).__init__(call_number1,title1,subjects1, notes1)
        self.author= author1;
        self.description=description1;
        self.publisher = publisher1;
        self.city = city1;
        self.year  = year1;
        self.series = series1;
    def display(self):
        print("Results")
        print("BOOK")
        super(Books,self).display()
        print(self.author)
        print(self.description)
        print(self.publisher)
        print(self.city)
        print(self.year)
        print(self.series)
    def compare_other(self,string_other):
        if (self.notes.find(string_other) != -1 or self.description.find(string_other) != -1 or self.year.find(string_other) != -1):
            return True
        return False
        
    



        
